### PR TITLE
XOL-4137 PicThrive app

### PR DESCRIPTION
**Pull request by jeetsang**

### CODE REVIEW

File: spec/support/jasmine.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "spec_dir": "spec",
+    "spec_files": ["**/*[sS]pec.js"],
+    "helpers": ["helpers/**/*.js"],
+    "stopSpecOnExpectationFailure": false,
+    "random": false
+}
```
Comment (anushr) [inline spec/support/jasmine.json]: Any reason why this is inside a `support` dir? I think leaving it in `spec` is nicer.

File: services/EmailTemplateService.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            Email: ${seller.email}<br>
+            Seller ID: ${seller.id}
+            </p>
+            <p>
+            Xola has unregistered the webhooks.
+            </p>
+            <p>
+            Thank you,<br>
+            Xola
+            </p>`;
```
[the remark this answers is not in the document]
Comment (jeetsang) [reply services/EmailTemplateService.js]: done

File: services/EmailTemplateService.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            Email: ${seller.email}<br>
+            Seller ID: ${seller.id}
+            </p>
+            <p>
+            If they already have an account with PicThrive, please ignore this request. If they do not have an account with PicThrive please create the account.<br>
+            The configuration for sending event details from Xola has already been completed.
+            </p>
+            <p>
+            Thank you,<br>
+            Xola
```
[the remark this answers is not in the document]
Comment (jeetsang) [reply services/EmailTemplateService.js]: done

File: config/default.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    }
+  },
+  "picthrive": {
+    "url": ""
+  },
+  "xola": {
+    "url": ""
+  },
+  "app": {
+    "port": 3002
```
[the remark this answers is not in the document]
Comment (jeetsang) [reply config/default.json]: changed

File: app.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+app.use('/', router);
+
+// catch 404 and forward to error handler
+app.use(function(req, res, next) {
+    next(createError(404));
+});
+
+app.use((err, req, res, next) => {
+    res.locals.message = err.message;
+    res.locals.error = req.app.get('env') === 'development' ? err : {};
```
[the remark this answers is not in the document]
Comment (jeetsang) [reply app.js]: removed custom error handler. Express has default error handler which automatically take care about the environment 
https://expressjs.com/en/guide/error-handling.html

File: README.md [hunk 6]
```diff
@@ -1,2 +1,20 @@
...
+1. Use `manifest\register_app.json` payload structure to register this app with **elrond**.
+
+##### Update Configs:
+2. Copy `config\default.json.` to `config\local.json` and update the configs in `config\local.json` accordingly. 
+ 
+
+##### Server setup:
+ ``` 
+ npm install
+ npm run start
```
Comment (anushr) [inline README.md]: Also specify how to start it with pm2 (for sandbox, production)

Comment (jeetsang) [reply README.md]: Done. I like pm2

File: README.md [hunk 7]
```diff
@@ -9,12 +9,20 @@ PicThrive Plugin for Xola App Store
...
+```
+npm install
+```
+2.Start the app with npm
+ ```
  npm run start
   ```
-##### run test:
- ``` 
+3.Start the app with pm2
```
Comment (anushr) [inline README.md]: This is confusion. You're saying that you should first start the app with `npm start` and then also start it with pm2.

Be clear. When should you use what? I clearly specified in the comment when to use pm2.

File: config/default.json [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    }
+  },
+  "picthrive": {
+    "url": ""
+  },
+  "xola": {
+    "url": ""
+  },
+  "app": {
+    "port": 8810
```
Comment (anushr) [inline config/default.json]: Actually, can you change the port to 11102? I discussed this with other and I think we want to keep app ports separate from Xola ports.

So the format will be 111xx for apps. Vinay's app got 11101, so you'll take 11102.

**Pull request merged**